shared_model/validators/query_validator.hpp

```diff
@@ -34,56 +34,75 @@ namespace shared_model {
     class QueryValidatorVisitor
         : public boost::static_visitor<ReasonsGroupType> {
      public:
-      QueryValidatorVisitor(const FieldValidator &validator = FieldValidator())
-          : validator_(validator) {}
-
       ReasonsGroupType operator()(
           const detail::PolymorphicWrapper<interface::GetAccount> &qry) const {
+        return this->operator()(*qry.get());
```

Comment:
Looks awful D: that one is a bit nicer, if I understand the logic
`(*this)(*qry)`

Reply:
Yes, looks bad. Your variant is worse imo, it is less readable.